dist/testbed-support/makefile.iotlab.single.inc.mk

```
@@ -57,7 +57,7 @@ endif
 IOTLAB_AUTH ?= $(HOME)/.iotlabrc
 IOTLAB_USER ?= $(shell cut -f1 -d: $(IOTLAB_AUTH))
 
-ifneq (0,$(shell command -v iotlab-experiment -h 2>&1 > /dev/null ; echo $$?))
+ifneq (0,$(shell command -v iotlab-experiment -h > /dev/null 2>&1; echo $$?))
```

```suggestion
ifneq (0,$(shell command -v iotlab-experiment -h > /dev/null echo $$?))
```
why not leave stderr in place?

---

Because the complexity of this is what keeps Make from being clever.

The whole line can be simplified to `ifeq (,$(shell command -v iotlab-experiment))`, but then this fails on every GNU Make < 4.3 (ie. every Ubuntu ever released) like what broke everything this morning.

---

Put more generally, what this PR is about is just about forcing out and fixing the particular bug of using `2>&1 >/dev/null`. There is a whole other issue about unifying the `which` and `command` calls in #16807, but I'd like to keep the focus here to fixing the sequence -- especially because eagerly fixing bycatch of `shell command` has been disastrous before.

---

So this will behave different without the stderr redirect?

---

It might; I'd need to dig up some of the old offending environments where make tries to forego the actual shell forking to evaluate it.

Once we're sure that no old make versions are around on users' systems any more (and we don't have a strict dependency there, but I'm holding off https://github.com/RIOT-OS/RIOT/pull/16807 for some more time), we can reevaluate all these, but until then, I'm trying to minimally fix the errors around `command`.